src/SwiftTaskProvider.ts

```diff
@@ -65,9 +70,10 @@ function createCleanTask(): vscode.Task {
  * Creates a {@link vscode.Task Task} to run an executable target.
  */
  function createBuildTasks(product: Product): vscode.Task[] {
+    const debugArguments = process.platform === 'win32' ? ['-Xlinker', '-debug:dwarf'] : [];
```

Comment:
If you are using codeview, this should be `-debug` rather than `-debug:dwarf`.  Also, you need to ensure that `-use-ld=link` is not passed in when using this argument as this is only valid with lld.

Reply:
(Oh, alternatively, and even better - ensure that `-use-ld=lld` is present for `-debug:dwarf`)

Reply:
I had to look up a few things here, so just to make sure I understand this correctly:

- Is CodeView the default on Windows (when just using `swift build` for example)?
- Should we always add `-debug` when creating a debug build using CodeView?
- What would be an appropriate name for this configuration setting? "Debug using DWARF"?

What should be the default, if we want to make sure things "just work" for new developers?

Reply:
> I had to look up a few things here, so just to make sure I understand this correctly:
> 
> * Is CodeView the default on Windows (when just using `swift build` for example)?

No, the default is no debug info :). `-g -debug-info-format=codeview` enables codeview, `-g` enables DWARF.

> * Should we always add `-debug` when creating a debug build using CodeView?

Yes, `-debug` should be specified always when building with CodeView debug info.

> * What would be an appropriate name for this configuration setting? "Debug using DWARF"?

That is what I would call it.  "Debug (DWARF)" and "Debug (CodeView)" is what I use.  But the using rather than the parenthetical is better for display purposes perhaps.

> What should be the default, if we want to make sure things "just work" for new developers?

My preference would be CodeView.  There are some bugs that we do need to sort out, but the same can be said for DWARF (and arguably more).  However, if you do use CodeView, you can deal with mixed-language projects (C/C++/Swift) and debug across system code and Swift code.  Additionally, I've found that lldb is still not particularly stable and causes trouble, and using DWARF mandates the use of lldb (at this point).